### PR TITLE
Proposing a new code layout & style.

### PR DESCRIPTION
**Pull request by andreihu**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,5 @@ else()
   message(WARNING "you can initialize you dependancies with 'conan install ..'")
 endif()
 
-add_subdirectory(Spck)
-add_subdirectory(Sandbox)
+add_subdirectory(src)
```
Comment (mudlee) [inline CMakeLists.txt:17]: I get an error, possibly for this line when I try to build it by running `./spck.sh -install && ./spck.sh -build`
```bash
CMake Error at CMakeLists.txt:17 (add_subdirectory):
  The source directory

    /Users/mudlee/Work/spck/src

  does not contain a CMakeLists.txt file.
```

File: src/sandbox/main.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include "sandbox.hpp"
```
Comment (mudlee) [inline src/sandbox/main.cpp:1]: Shouldn't it be?
```c++
#include <sandbox.hpp>
```

File: src/spck/spck/api.hpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#ifdef SPCK_PLATFORM_WINDOWS
```
Comment (mudlee) [inline src/spck/spck/api.hpp:3]: Wouldn't look this better in this way?
```c++
#ifdef SPCK_PLATFORM_WINDOWS
    #ifdef SPCK_BUILD_DLL
        #define SPCK_API __declspec(dllexport)
    #else
        #define SPCK_API __declspec(dllimport)
#endif
#else
    #define SPCK_API
#endif
```

File: src/spck/spck/application.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <spck/api.hpp>
+#include <spck/log.hpp>
+#include <spck/messaging/application_event.hpp>
+#include <spck/messaging/event.hpp>
+#include <spck/messaging/key_event.hpp>
+
+namespace spck {
+class SPCK_API application {
```
Comment (mudlee) [inline src/spck/spck/application.hpp:10]: These three lines (namespace, class, public) after each and other just makes it hard to read.
Can we use a convention that at least between namespace and class you have to have an empty line?
```c++
namespace spck {

class SPCK_API application {
}

}

File: src/spck/spck/messaging/application_event.hpp [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include "event.hpp"
```
Comment (mudlee) [inline src/spck/spck/messaging/application_event.hpp:3]: `#include <event.hpp>` ?

File: src/spck/spck/messaging/key_event.hpp [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "event.hpp"
```
Comment (mudlee) [inline src/spck/spck/messaging/key_event.hpp:3]: `#include <event.hpp>`?

File: src/spck/spck/spck.hpp [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+#include "application.hpp"
```
Comment (mudlee) [inline src/spck/spck/spck.hpp:3]: Shouldn't we use <>?

File: src/spck/spck/window/window.cpp [hunk 8]
```diff
@@ -0,0 +1,69 @@
+#include "window.hpp"
```
Comment (mudlee) [inline src/spck/spck/window/window.cpp:1]: Shouldn't we use <>?

**Review by mudlee — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (andreihu) [reply CMakeLists.txt:17]: If the CMakeLists.txt file is there, then you might want to try to delete the build folder and start from scratch.

File: src/sandbox/main.cpp (hunk 2 above)
Comment (andreihu) [reply src/sandbox/main.cpp:1]: That'll become an include error.

File: src/spck/spck/api.hpp (hunk 3 above)
Comment (andreihu) [reply src/spck/spck/api.hpp:3]: It was clang-format which is formatting down to this style. I'll try to improve it.

File: src/spck/spck/application.hpp (hunk 4 above)
Comment (andreihu) [reply src/spck/spck/application.hpp:10]: I'll try to improve it.

File: src/spck/spck/messaging/application_event.hpp (hunk 5 above)
Comment (andreihu) [reply src/spck/spck/messaging/application_event.hpp:3]: That's gonna end up with a missing header error.

File: src/spck/spck/messaging/key_event.hpp (hunk 6 above)
Comment (andreihu) [reply src/spck/spck/messaging/key_event.hpp:3]: That's gonna end up with a missing header error.

File: src/spck/spck/spck.hpp (hunk 7 above)
Comment (andreihu) [reply src/spck/spck/spck.hpp:3]: That's gonna end up with a missing header error.

File: src/spck/spck/window/window.cpp (hunk 8 above)
Comment (andreihu) [reply src/spck/spck/window/window.cpp:1]: That's gonna end up with a missing header error.

**Review by mudlee — APPROVED** [review]

**Pull request merged**